search/src/main/java/artic/edu/search/DefaultSearchSuggestionsViewModel.kt

```diff
@@ -0,0 +1,165 @@
+import io.reactivex.rxkotlin.Observables
+import io.reactivex.subjects.BehaviorSubject
+import io.reactivex.subjects.Subject
+import javax.inject.Inject
+
+
+/**
+ * @author Sameer Dhakal (Fuzz)
+ */
+class DefaultSearchSuggestionsViewModel @Inject constructor(searchSuggestionsDao: ArticSearchObjectDao,
+                                                            objectDao: ArticObjectDao
+) : BaseViewModel() {
+
+    private val suggestedKeywords: Subject<List<TextCellViewModel>> = BehaviorSubject.create()
+    private val suggestedArtworks: Subject<List<CircularCellViewModel>> = BehaviorSubject.create()
+    val cells: Subject<List<SearchBaseCellViewModel>> = BehaviorSubject.create()
+
+    private val cellViewHolders: List<SearchBaseCellViewModel>?
+        get() {
+            val viewHolders = (cells as BehaviorSubject)
+            return if (viewHolders.hasValue()) {
+                viewHolders.value
+            } else {
+                null
+            }
+        }
+
+    private fun getAmenitiesVieModels(): List<SearchBaseCellViewModel> {
+        return listOf(
+                AmenitiesCellViewModel(R.drawable.ic_icon_amenity_map_restaurant),
+                AmenitiesCellViewModel(R.drawable.ic_icon_amenity_map_lounge),
+                AmenitiesCellViewModel(R.drawable.ic_icon_amenity_map_shop),
+                AmenitiesCellViewModel(R.drawable.ic_icon_amenity_map_restroom))
+    }
+
+    init {
+
+
+        searchSuggestionsDao.getDataObject()
+                .toObservable()
+                .map { suggestedSearchOptions ->
+                    val list = mutableListOf<TextCellViewModel>()
+                    suggestedSearchOptions.searchStrings.forEach { keyword ->
+                        list.add(TextCellViewModel(SearchViewComponent.SuggestedKeyword(keyword)))
+                    }
+                    return@map list
+                }
+                .bindTo(suggestedKeywords)
+                .disposedBy(disposeBag)
+
+        searchSuggestionsDao.getDataObject()
+                .toObservable()
+                .map { suggestedSearchOptions -> suggestedSearchOptions.searchObjects }
+                .flatMap { idsList ->
+                    objectDao.getObjectsByIdList(idsList).toObservable()
+                }
+                .map { objects ->
+                    val list = mutableListOf<CircularCellViewModel>()
+                    objects.forEach { artwork ->
+                        list.add(CircularCellViewModel(SearchViewComponent.Artwork(artwork)))
+                    }
+                    return@map list
+                }
+                .bindTo(suggestedArtworks)
+                .disposedBy(disposeBag)
+
+        Observables.combineLatest(suggestedArtworks, suggestedKeywords)
+        { artworks, keywords ->
+            mutableListOf<SearchBaseCellViewModel>()
+                    .apply {
+                        add(0, HeaderCellViewModel(SearchViewComponent.Header("Suggested")))
```

Comment:
Suggested is not localized here. use a resource Id and look it up at binding time.

Reply:
VM now emits string resource id.